File: Arrays/Two_Number_Sum/Two_Number_Sum.js
//import  {binarySearch} from "../../Famous Algos/Binary Search/BinarySearch";
const binarySearch = require("../../Famous Algos/Binary Search/BinarySearch");
console.log(binarySearch);

const twoNumberSum = (array, targetSum) => {
  const sortedArray = array.sort((a, b) => a - b); // (merge || quick) sort
  let result = [];
  const midIndex = Math.floor(sortedArray.length / 2);

  for (let i = 0; i < midIndex; i++) {
    const searchNum = targetSum - sortedArray[i];
    const theBinarySearch = binarySearch(sortedArray, searchNum);
    if (theBinarySearch[0] === true) {
      result.push([sortedArray[i], searchNum]);
    }
  }

  return result;
};

console.log(twoNumberSum([1, 2, 4, 5, 7, 6, 9, 10, 11], 11));

// 0 complexity  = n/2 * nlogn

File: Arrays/1. Two_Number_Sum/Two_Number_Sum.py
import math

def twoNumberSum (array:list, targetSum:int ):
    array.sort()
    sortedArr = array
    midPos = math.floor(len(sortedArr)/2)
    resultArr = []
    
    for x in range(0,midPos):
        queryNum = targetSum - sortedArr[x]
        if queryNum in sortedArr:
            resultArr.append([sortedArr[x],queryNum]) 
    return resultArr


print(twoNumberSum([1, 2, 4, 5, 7, 6, 9, 10, 11], 11))